src/igvfd/types/image.py

```diff
@@ -60,3 +60,15 @@ def thumb_nail(self, request, attachment):
     )
     def download_url(self, request, attachment):
         return self.jsonld_id(request) + attachment['href']
+
+    @calculated_property(
+        condition='caption',
+        schema={
+            'title': 'Summary',
+            'type': 'string',
+            'notSubmittable': True,
+        }
+    )
+    def summary(self, caption):
+        if caption:
+            return caption
```

Comment:
In both cases, I would add an else statement to return UUID if there is no caption or title. I think you may also need to write `caption=None` or `title=None` in the function args in case the property is not present since it's not required.

Reply:
fixed.

Reply:
Could you also add to it to return `uuid` if no caption?

And sorry, I made a mistake about Page (will comment below)